modules/programs/vim.nix

```diff
@@ -135,7 +144,7 @@ in {
       ${cfg.extraConfig}
     '';
 
-    vim = pkgs.vim_configurable.customize {
+    vim = cfg.package.customize {
```

Comment:
Maybe add an assert that `customize` is available in the package set in `programs.vim.package` so users don't use unwrapped Vim by mistake :thinking: ?

Reply:
Something like:

```nix
{
  assertions = [{
    assertion = (config.programs.package ? customize);
    message = "`programs.vim.package' must be a `pkgs.vim_configurable' derivation";
  }];
}

Reply:
Discussed here: https://github.com/nix-community/home-manager/pull/1745#discussion_r563995652

Reply:
actually I think it would be best for vim to be install an unwrapped vim (like is done for neovim) and generate a ~/.vimrc. Wrapping vim has sideeffects regarding vim's behavior.

Reply:
Using `~/.vimrc` does make more sense to me, but it does seem that neovim uses a wrapped package still :thinking: ? https://github.com/nix-community/home-manager/blob/master/modules/programs/neovim.nix#L265-L269

Is there some reason for it @teto? If yes, does the same reason apply here or not?

Reply:
Right I was not precise, wrapping with `-u` has side effects .

Reply:
Regarding the assertion, there is no such check in the neovim module for configure, so this would divert from that approach a bit. Fine by me certainly, just wan to make sure this is expected.

Reply:
I suppose that's because in neovim you can always pass the unwrapped version since all config is done at runtime contrary to vim.